ci/do_ci.ps1

```diff
@@ -4,17 +4,41 @@ trap { $host.SetShouldExit(1) }
 . "$PSScriptRoot\build_setup.ps1"
 Write-Host "building using $env:NUM_CPUS CPUs"
 
-function bazel_debug_binary_build() {
+function bazel_binary_build($type) {
   echo "Building..."
-  pushd "$env:ENVOY_SRCDIR"
-    bazel  $env:BAZEL_BASE_OPTIONS.Split(" ") build $env:BAZEL_BUILD_OPTIONS.Split(" ") -c dbg "//source/exe:envoy-static"
-    $exit = $LASTEXITCODE
-    if ($exit -ne 0) {
-      popd
-      exit $exit
-    }
-  popd
+  bazel  $env:BAZEL_BASE_OPTIONS.Split(" ") build $env:BAZEL_BUILD_OPTIONS.Split(" ") -c $type "//source/exe:envoy-static"
+  $exit = $LASTEXITCODE
+  if ($exit -ne 0) {
+    exit $exit
+  }
 }
 
-echo "bazel debug build..."
-bazel_debug_binary_build
+function bazel_test($type, $test) {
+  if ($test -ne "") {
+    bazel $env:BAZEL_BASE_OPTIONS.Split(" ") test $env:BAZEL_TEST_OPTIONS.Split(" ") -c $type $test
+  } else {
+    echo "running windows tests"
+    bazel $env:BAZEL_BASE_OPTIONS.Split(" ") test $env:BAZEL_TEST_OPTIONS.Split(" ") -c $type "//test:windows_tests"
+  }
+  exit $LASTEXITCODE
+}
+
+$action = $args[0]
+$test = $args[1]
+
+switch ($action) {
+  "bazel.debug" {
+    echo "bazel debug build with tests..."
+    # TODO(sesmith177) uncomment when Envoy builds on Windows
+    #bazel_binary_build "dbg"
+    bazel_test "dbg" "$test"
+  }
+  "bazel.debug.server_only" {
+    echo "bazel debug build..."
+    bazel_binary_build "dbg"
+  }
+  default {
+    echo "unknown action: $action"
+    exit 1
+  }
```

Comment:
Can you split out these "build goodness for Windows" changes to a separate PR? Thanks.

Reply:
Sure, we'll do that